algorithms/kernel/k_nearest_neighbors/kdtree_knn_classification_predict_dense_default_batch_impl.i

```diff
@@ -376,23 +307,16 @@ void KNNClassificationPredictKernel<algorithmFpType, defaultDense, cpu>::findNea
                 DAAL_PREFETCH_READ_T0(nx);
                 DAAL_PREFETCH_READ_T0(nx + 16);
 
-                for (i = 0; i < end - start; ++i)
-                {
-                    distance[i] += (query[j - 1] - dx[i]) * (query[j - 1] - dx[i]);
-                }
+                distance_comp<cpu, algorithmFpType>(dx, query[j - 1], distance, length);
```

Comment:
Do you specified algorithmFpType as a template argument explicitly to perform some conversion?

Reply:
No, it doesn't require any conversion and it this type can be deduced by compiler. But explicit template specification affects neither performance nor extensibility, right?